dae_testing/homework_example/HW8_3a.py

```diff
@@ -39,24 +72,55 @@ def _x3dot(m, t):
         return m.dx3dt[t] == m.x1[t]**2 + m.x2[t]**2 + 0.005*m.u[t]**2
     m.x3dot = Constraint(m.t, rule=_x3dot)
 
-    def _init(m):
-        yield m.x1[0] == 0
-        yield m.x2[0] == -1
-        yield m.x3[0] == 0
-    m.init_conditions = ConstraintList(rule=_init)
-
-    return m
+#Constraint_definitions_old doesn't ignore constraints at 
+#non-collocation discretization points
 
 
 def discretize_model(
         m,
         method="dae.collocation",
         scheme="LAGRANGE-RADAU",
-        nfe=10,
-        ncp=5,
+        nfe= 2,
+        ncp=3,
         ):
     discretizer = TransformationFactory(method)
     discretizer.apply_to(m, nfe=nfe, ncp=ncp, scheme=scheme)
+    
+
+def discretization_points(m):
+    return m.t.get_finite_elements(), \
+        m.t.get_discretization_info()['tau_points']
+
+#m.t.get_finite_elements = These are all discretization points 
+#m.t.tau_pts = These are roots of polynomials including 0 
+#as an additional element
+
+
+def convert_taupts_to_cp(m, nfe):
+    h = m.tf/nfe
+    cp = []
+    for i in m.disc_pts[:-1] :
+        for j in m.col_roots[1:]:
+            cp.append(i + h*j)
+    return cp
+
+#These convert the polynomial roots to collocation points between 
+#each of the finite element boundaries.

```

Comment:
This seems to assume that collocation points are evenly spaced? Also, it relies on the `tf`, `disc_pts`, and `col_roots` attributes being attached to the model. This makes it less general than it could be. I would just take `disc_pts` and `col_roots` as arguments, and do the calculation from there. (`tf` is just `disc_pts[-1]` and `nfe` is just `len(disc_pts)-1`.)

Reply:
Also, same comment as above on docstring and code location.